crates/searchers/openapi.yml

```diff
@@ -0,0 +1,630 @@
+          $ref: "#/components/schemas/Signature"
+
+    Trade:
+      description: |
+        A trade for a CoW Protocol order included in a solution.
+      type: object
+      properties:
+        order:
+          description: |
+            The order to execute in a solution. Can either be a reference by UID
+            to an order included in the auction, or a JIT liquidity order.
+          oneOf:
+            - $ref: "#/components/schemas/OrderUid"
+            - $ref: "#/components/schemas/JitOrder"
+        executedAmount:
+          description: |
+            The amount of the order that was executed. This is denoted in
+            "sellToken" for sell orders, and "buyToken" for buy orders.
+          allOf:
+            - $ref: "#/components/schemas/TokenAmount"
+
+    LiquidityInteraction:
+      description: |
+        Interaction representing the execution of liquidity that was passed in
+        with the auction.
+      type: object
+      required:
+        - kind
+        - index
+        - inputToken
+        - outputToken
+        - inputAmount
+        - outputAmount
+      properties:
+        kind:
+          type: string
+          enum: [liquidity]
+        id:
+          description: |
+            The ID of executed liquidity provided in the auction input.
+          type: number
+        inputToken:
+          $ref: "#/components/schemas/Token"
+        outputToken:
+          $ref: "#/components/schemas/Token"
+        inputAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        outputAmount:
+          $ref: "#/components/schemas/TokenAmount"
+
+    Approval:
+      description: |
+        A requested approval from the settlement contract to some spender.
+      type: object
+      required:
+        - kind
+        - token
+        - spender
+        - minAmount
+      properties:
+        kind:
+          type: string
+          enum: [approval]
+        token:
+          $ref: "#/components/schemas/Token"
+        spender:
+          $ref: "#/components/schemas/Address"
+        minAmount:
+          $ref: "#/components/schemas/TokenAmount"
+
+    CustomInteraction:
+      description: |
+        A searcher-specified custom interaction to be included in the final
+        settlement.
+      type: object
+      required:
+        - kind
+        - target
+        - value
+        - callData
+        - inputs
+        - outputs
+      properties:
+        kind:
+          type: string
+          enum: [custom]
+        target:
+          $ref: "#/components/schemas/Address"
+        value:
+          $ref: "#/components/schemas/TokenAmount"
+        callData:
+          description: |
+            The EVM calldata bytes.
+          type: string
+          example: "0x01020304"
+        approvals:
+          description: |
+            ERC20 approvals that are required for this custom interaction.
+          type: array
+          items:
+            $ref: "#/components/schemas/Approval"
```

Comment:
@ennmichael - this was an "ah-ha" moment for me, but approvals are only needed for custom interactions. For liquidity interactions, we know how to encode them and what approvals are needed. This is great for two reasons:
- When we "internalize" interactions, we don't need to "find" the equivalent approval and also remove from the settlement
- We don't need an additional field in the `Solution` type.

Reply:
We do, but the approvals for liquidity interactions don't need to be specified in the solution (as they are implicit).